nvm.sh

```diff
@@ -2045,11 +2088,9 @@ nvm_install_source() {
 
   TARBALL="$(PROGRESS_BAR="${PROGRESS_BAR}" nvm_download_artifact "${FLAVOR}" source "${TYPE}" "${VERSION}" | command tail -1)" && \
   [ -f "${TARBALL}" ] && \
-  TMPDIR="$(dirname "${TARBALL}")/files" && \
+  TMPDIR="$(nvm_extract_tar_artifact "${TARBALL}" "${NVM_OS}" | command tail -1)" && \
   if ! (
     # shellcheck disable=SC2086
```

Comment:
I saw that this line might be necessary while running `tar`, since line 1950:
```bash
command ${tar} -x${tar_compression_flag}f ${TARBALL} -C ${TMPDIR} ${TAR_COMMAND_SUFFIX}
```
errors out when `${TAR_COMMAND_SUFFIX}` is empty resulting in an extra parameter to `tar` and the following error:
```bash
tar: Empty pattern string
```

So I'm moving it from here to line `1950`, ok?